core/iwasm/common/wasm_runtime_common.c

```diff
@@ -1350,12 +1350,14 @@ wasm_runtime_load_ex(uint8 *buf, uint32 size, const LoadArgs *args,
                                           true,
 #endif
                                           args, error_buf, error_buf_size);
+        ((WASMModule *)module_common)->is_binary_freeable = true;
```

Comment:
Should be `((WASMModule *)module_common)->is_binary_freeable = args.wasm_binary_freeable;`

Reply:
And should also modify `wasm_runtime_load_from_sections` in this file, set `((WASMModule *)module_common)->is_binary_freeable =true` and `((AOTModule *)module_common)->is_binary_freeable = true`